### PR TITLE
[FrameworkBundle] Move debug configuration to PHP

### PR DESCRIPTION
**Pull request by TuxBoy**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | Ref #37186
| License       | MIT

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/debug.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+use Symfony\Component\HttpKernel\Controller\ArgumentResolver\NotTaggedControllerValueResolver;
+use Symfony\Component\HttpKernel\Controller\TraceableArgumentResolver;
+use Symfony\Component\HttpKernel\Controller\TraceableControllerResolver;
+use Symfony\Component\HttpKernel\Debug\TraceableEventDispatcher;
+
+return static function (ContainerConfigurator $container) {
+    $container->services()
+        ->set('debug.event_dispatcher', TraceableEventDispatcher::class)
+        ->decorate('event_dispatcher')
```
Comment (yceruto) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/debug.php:22]: We need an extra indent after each `->set()`, check CS style in this PR: https://github.com/symfony/symfony/pull/36778/files

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/debug.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        ->decorate('argument_resolver')
+        ->args([
+            service('debug.argument_resolver.inner'),
+            service('debug.stopwatch')
+        ])
+
+        ->set('argument_resolver.not_tagged_controller', NotTaggedControllerValueResolver::class)
+        ->tag('controller.argument_value_resolver', ['priority' => -200])
+        ->args([abstract_arg('Controller argument, set in FrameworkExtension')])
+
```
Comment (yceruto) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/debug.php:49]: please remove this blank line

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/debug_prod.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Loader\Configurator;
+
+use Symfony\Component\HttpKernel\Debug\FileLinkFormatter;
+use Symfony\Component\HttpKernel\EventListener\DebugHandlersListener;
+
+return static function (ContainerConfigurator $container) {
+    $kernelDebugServiceId = 'kernel.debug';
```
Comment (yceruto) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/debug_prod.php:18]: To be consistent with other PR, let's use the value `kernel.debug` directly for both params.

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/debug.php [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+use Symfony\Component\HttpKernel\Controller\TraceableArgumentResolver;
+use Symfony\Component\HttpKernel\Controller\TraceableControllerResolver;
+use Symfony\Component\HttpKernel\Debug\TraceableEventDispatcher;
+
+return static function (ContainerConfigurator $container) {
+    $container->services()
+        ->set('debug.event_dispatcher', TraceableEventDispatcher::class)
+            ->decorate('event_dispatcher')
+            ->tag('monolog.logger', ['channel' => 'event'])
+            ->tag('kernel.reset', ['method' => 'reset'])
```
Comment (yceruto) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/debug.php:24]: let's place tag after args and calls, ref https://github.com/symfony/symfony/pull/37211#issuecomment-642639486

**Comment by yceruto** [comment]
please check the fabbot report, there are other issue CS to solve: 
https://fabbot.io/report/symfony/symfony/37212/d75bd684374658538d9f017173d1fa1ef3b9b6bc

**Review by yceruto — APPROVED** [review]
LGTM, thanks!

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @TuxBoy.

**Pull request merged**